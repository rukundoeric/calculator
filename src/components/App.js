import logo from '../assets/images/logo.svg';
import '../assets/css/style.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          calculator app
        </p>
      </header>
    </div>
  );
}

export default App;
