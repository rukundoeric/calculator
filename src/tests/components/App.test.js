import '@testing-library/jest-dom';
import { render, fireEvent, cleanup } from '@testing-library/react';
import App from '../../components/App';

afterEach(cleanup);

it('Should match snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

it('Should display 0 by default', () => {
  const { getByTestId } = render(<App />);
  expect(getByTestId('display_res')).toHaveTextContent('0');
});

it('Should display number from which the buttons you clicked on.', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_3'));
  fireEvent.click(getByTestId('button_6'));
  expect(getByTestId('display_res')).toHaveTextContent('436');
});

it('Should display calculation path', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_+'));
  fireEvent.click(getByTestId('button_6'));
  expect(getByTestId('display_res')).toHaveTextContent('4+6');
});

it('Should display persentage of entered number', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_0'));
  fireEvent.click(getByTestId('button_%'));
  expect(getByTestId('display_res')).toHaveTextContent('0.4');
});

it('Should display the result of operation:+/- ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_+/-'));
  expect(getByTestId('display_res')).toHaveTextContent('-42');
});

it('Should display the result of operation:+ ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_+'));
  fireEvent.click(getByTestId('button_9'));
  fireEvent.click(getByTestId('button_='));
  expect(getByTestId('display_res')).toHaveTextContent('51');
});

it('Should display the result of operation:- ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_-'));
  fireEvent.click(getByTestId('button_9'));
  fireEvent.click(getByTestId('button_='));
  expect(getByTestId('display_res')).toHaveTextContent('33');
});

it('Should display the result of operation:* ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_X'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_='));
  expect(getByTestId('display_res')).toHaveTextContent('84');
});

it('Should display the result of operation:รท ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_รท'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_='));
  expect(getByTestId('display_res')).toHaveTextContent('21');
});

it('Should display the result of operation:รท ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_รท'));
  fireEvent.click(getByTestId('button_0'));
  fireEvent.click(getByTestId('button_='));
  expect(getByTestId('display_res')).toHaveTextContent('INFINITY');
});

it('Should clear everything and display 0 ', () => {
  const { getByTestId } = render(<App />);
  fireEvent.click(getByTestId('button_4'));
  fireEvent.click(getByTestId('button_2'));
  fireEvent.click(getByTestId('button_5'));
  expect(getByTestId('display_res')).toHaveTextContent('425');
  fireEvent.click(getByTestId('button_AC'));
  expect(getByTestId('display_res')).toHaveTextContent('0');
});
