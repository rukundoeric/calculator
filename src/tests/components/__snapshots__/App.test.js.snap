// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="calc-container"
    >
      <div
        class="calc-content"
      >
        <table>
          <thead
            height="100px"
            id="dispay-h"
          >
            <tr>
              <td
                colspan="4"
              >
                <div
                  class="res"
                  data-testid="display_res"
                >
                  0
                </div>
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                colspan=""
              >
                <button
                  class="contrl"
                  data-testid="button_AC"
                  datavalue="AC"
                  type="button"
                >
                  AC
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="contrl"
                  data-testid="button_+/-"
                  datavalue="+/-"
                  type="button"
                >
                  +/-
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="contrl"
                  data-testid="button_%"
                  datavalue="%"
                  type="button"
                >
                  %
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="ops"
                  data-testid="button_รท"
                  datavalue="รท"
                  type="button"
                >
                  รท
                </button>
              </td>
            </tr>
            <tr>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_7"
                  datavalue="7"
                  type="button"
                >
                  7
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_8"
                  datavalue="8"
                  type="button"
                >
                  8
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_9"
                  datavalue="9"
                  type="button"
                >
                  9
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="ops"
                  data-testid="button_X"
                  datavalue="X"
                  type="button"
                >
                  X
                </button>
              </td>
            </tr>
            <tr>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_4"
                  datavalue="4"
                  type="button"
                >
                  4
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_5"
                  datavalue="5"
                  type="button"
                >
                  5
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_6"
                  datavalue="6"
                  type="button"
                >
                  6
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="ops"
                  data-testid="button_-"
                  datavalue="-"
                  type="button"
                >
                  -
                </button>
              </td>
            </tr>
            <tr>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_1"
                  datavalue="1"
                  type="button"
                >
                  1
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_2"
                  datavalue="2"
                  type="button"
                >
                  2
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_3"
                  datavalue="3"
                  type="button"
                >
                  3
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="ops"
                  data-testid="button_+"
                  datavalue="+"
                  type="button"
                >
                  +
                </button>
              </td>
            </tr>
            <tr>
              <td
                colspan="2"
              >
                <button
                  class="nums"
                  data-testid="button_0"
                  datavalue="0"
                  type="button"
                >
                  0
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="nums"
                  data-testid="button_."
                  datavalue="."
                  type="button"
                >
                  .
                </button>
              </td>
              <td
                colspan=""
              >
                <button
                  class="ops"
                  data-testid="button_="
                  datavalue="="
                  type="button"
                >
                  =
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
